### ISSUE TITLE
API/DEPR: infer_freq/inferred_freq return an offset object

### ISSUE THREAD

**Issue by jbrockmendel** [pandas-dev/pandas#55504]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
NA
```

### Issue Description

pd.tseries.frequencies.infer_freq, DatetimeIndex.inferred_freq, TimedeltaIndex.inferred_freq all return a string that we then have to cast to an offset object.  it would be more ergonomic to just return an offset to begin with.

### Expected Behavior

NA

### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

**Comment by rhshadrach** [comment]
Agreed, and this would also makes `inferred_freq` / `freq` both return offsets.
